server/src/main/java/com/thoughtworks/go/server/service/support/toggle/FeatureToggleService.java

```diff
@@ -42,25 +42,56 @@ public FeatureToggleService(FeatureToggleRepository repository, GoCache goCache)
     }
 
     public boolean isToggleOn(String key) {
-        FeatureToggle toggle = allToggles().find(key);
+        // handle any key, really.
+        FeatureToggle toggle = userToggles().find(key);
+        if (toggle != null) {
+            return toggle.isOn();
+        }
+
+        toggle = allToggles().find(key);
```

Comment:
It may not be obvious, but `allToggles()` is _not really_ all toggles. It filters out toggles that are not known to `available.toggles`. Because we _may_ have dynamically generated toggle keys for route toggles, we need to account for these in case users set them in `<configDir>/go.feature.toggles`.

Thus, we first search `userToggles()` alone before defaulting to `allToggles()`.

Reply:
@marques-work I realized we cannot change the value for a dynamic toggle key. If you try to update the dynamic toggle key `config_repo_v4` using the feature_toggles API it fails.
The reason for the failure is this check - https://github.com/gocd/gocd/blob/3423ef97167847e1f57af32ee157e405a205ebcf/server/src/main/java/com/thoughtworks/go/server/service/support/toggle/FeatureToggleService.java#L102